### PR TITLE
chore: updates CRS tests to CRS 4.0.0-rc2

### PR DESCRIPTION
**Pull request by M4tteoP**
- Updates CRS tests to CRS 4.0.0-rc2
   - `/post` has been added to the emulated httpbin behavior, being used for a couple of tests to match the response body.
   - Notably: `934131-5` and `934131-7` are failing. This is happening because:
         - we are urldecoding the cookie key values before storing them
         - The test comes with a `+` in the payload: `Yy9vbmVycm9yJTNkYWxlcnQoMSk+Cg==` (snippet of payload). The rule transformations are: `t:none,t:urlDecodeUni,t:jsDecode,t:base64Decode,t:urlDecodeUni,t:jsDecode`. Being the urldecode before the base64decode, the + is translated into a space, making the base64 fail. Further investigations (and likely comparisons with ModSec runs) are needed.

### CODE REVIEW

**Comment by jcchavezs** [comment]
cc @anuraaga @fzipi @jptosso

File: testing/coreruleset/.ftw.yml [hunk 1]
```diff
@@ -4,11 +4,10 @@ testoverride:
...
     920100-8: 'Go/http allows a colon in the path. Test expects status 400 or 403 (Apache behaviour)'
-    920170-3: 'HEAD request with data. Go/http does not allow it - 400 Bad Request'
     920270-4: 'Rule works, log contains 920270. Test expects status 400 (Apache behaviour)'
     920272-5: 'Rule works, log contains 920272. Test expects status 400 (Apache behaviour)'
     920290-1: 'Rule works, log contains 920290. Test expects status 400 (Apache behaviour)'
-    920420-8: 'HEAD request with data. Go/http does not allow it - 400 Bad Request'
-    920430-5: 'Test has expect_error, Go/http and Envoy return 400'
     920430-8: 'Go/http does no allow HTTP/3.0 - 505 HTTP Version Not Supported'
     932200-13: 'wip'
+    934131-5: 'TODO: check why we are failing to deobfuscate these payloads'
```
Comment (anuraaga) [inline testing/coreruleset/.ftw.yml:12]: Can you add the notes from the PR description here?

**Comment by jcchavezs** [comment]
Can we finish this @M4tteoP ?

**Comment by M4tteoP** [comment]
> - The test comes with a + in the payload: Yy9vbmVycm9yJTNkYWxlcnQoMSk+Cg== (snippet of payload). The rule transformations are: t:none,t:urlDecodeUni,t:jsDecode,t:base64Decode,t:urlDecodeUni,t:jsDecode. Being the urldecode before the base64decode, the + is translated into a space, making the base64 fail. Further investigations (and likely comparisons with ModSec runs) are needed.

Further investigations led to:
- https://github.com/corazawaf/coraza/issues/920
- https://github.com/corazawaf/coraza/issues/926

I added the links to the description of the ignored test and also mentioned that the rule that is tested by the two excluded tests is actually quite debated CRS side being very specific to a few payloads. More context: https://github.com/coreruleset/coreruleset/issues/3376

File: http/middleware_test.go [hunk 2]
```diff
@@ -101,8 +101,7 @@ SecRule &REQUEST_HEADERS:Transfer-Encoding "!@eq 0" "id:1,phase:1,deny"
 	}
 	if it == nil {
 		t.Fatal("Expected interruption")
-	}
-	if it.RuleID != 1 {
+	} else if it.RuleID != 1 {
```
Comment (M4tteoP) [inline http/middleware_test.go:104]: The linter was complaining with `SA5011: possible nil pointer dereference (staticcheck)`

File: testing/coreruleset/.ftw.yml [hunk 3]
```diff
@@ -9,5 +9,3 @@ testoverride:
     920290-1: 'Rule works, log contains 920290. Test expects status 400 (Apache behaviour)'
     920430-8: 'Go/http does no allow HTTP/3.0 - 505 HTTP Version Not Supported'
     932200-13: 'wip'
-    934131-5: 't:base64decode too strict https://github.com/corazawaf/coraza/issues/926 and https://github.com/corazawaf/coraza/issues/920. Rule itself is debated CRS side: https://github.com/coreruleset/coreruleset/issues/3376'
-    934131-7: 't:base64decode too strict https://github.com/corazawaf/coraza/issues/926 and https://github.com/corazawaf/coraza/issues/920. Rule itself is debated CRS side: https://github.com/coreruleset/coreruleset/issues/3376'
```
Comment (M4tteoP) [inline testing/coreruleset/.ftw.yml]: Tests `934131-5` and `934131-7` are now fixed

**Review by anuraaga — APPROVED** [review]

**Pull request merged**